### DISCUSSION TITLE
unable to download the asset using curl or wget command

### DISCUSSION THREAD

**Discussion by ThotadurgamaniTR** [community/community#67271] [Actions]
### Select Topic Area

General

### Body

iam able to download the asset using direct URL but unable to download it using curl command i provided PAT token also coudl you please help me iwth that

**Comment by d4kw1n** [comment]
can you describe in more detail?

**Comment by ThotadurgamaniTR** [comment]
This email is from a contractor that provides managed services to 3rd parties, including Thomson Reuters.
Hi Tri Pham Minh,

I have .zip files in my releases with tag_name=”tag4” repository_name “Boomi_TEMP” so I can able to get the release assests using “curl -L -H 'Accept: application/vnd.github+json' -H 'Authorization: token ***' -H 'X-GitHub-Api-Version: 2022-11-28' https://api.github.com/repos/ThotadurgamaniTR/Boomi_TEMP/releases/latest”

Above command given me the browser download URL example 'https://github.com/ThotadurgamaniTR/Boomi_TEMP/releases/download/tag4/consolidated_41a23303-d5ff-40b5-8a07-4c86a1c846c2.zip
I can able to download the assets using direct link but unable to download the asset using curl or wget command . but .zip file is not downloading. Also used the PAT token for authentication . don’t know the reaons it is giving me


++ basename https://github.com/ThotadurgamaniTR/Boomi_TEMP/releases/download/tag4/consolidated_41a23303-d5ff-40b5-8a07-4c86a1c846c2.zip
+ file_name=consolidated_41a23303-d5ff-40b5-8a07-4c86a1c846c2.zip
+ file_path=/home/runner/work/Boomi_TEMP/Boomi_TEMP/QA/consolidated_41a23303-d5ff-40b5-8a07-4c86a1c846c2.zip
+ wget '--header=Authorization: token ***' -O /home/runner/work/Boomi_TEMP/Boomi_TEMP/QA/consolidated_41a23303-d5ff-40b5-8a07-4c86a1c846c2.zip https://github.com/ThotadurgamaniTR/Boomi_TEMP/releases/download/tag4/consolidated_41a23303-d5ff-40b5-8a07-4c86a1c846c2.zip
--2023-09-15 18:40:28--  https://github.com/ThotadurgamaniTR/Boomi_TEMP/releases/download/tag4/consolidated_41a23303-d5ff-40b5-8a07-4c86a1c846c2.zip
Resolving github.com (github.com)... 192.30.255.112
Connecting to github.com (github.com)|192.30.255.112|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2023-09-15 18:40:29 ERROR 404: Not Found.

Error: Process completed with exit code 8.


Give me the reason why its noy downloading.

Thanks & regards,
Durga
From: Tri Pham Minh ***@***.***>
Sent: Saturday, September 16, 2023 1:18 PM
To: community/community ***@***.***>
Cc: Thota, Durgamani (Extranet) ***@***.***>; Author ***@***.***>
Subject: [EXT] Re: [community/community] unable to download the asset using curl or wget command (Discussion #67271)

External Email: Use caution with links and attachments.


can you describe in more detail?

**Comment by d4kw1n** [comment]
You can use this link to try downloading: "https://raw.githubusercontent.com/ThotadurgamaniTR/Boomi_TEMP/releases/download/tag4/consolidated_41a23303-d5ff-40b5-8a07-4c86a1c846c2.zip"
or command will be:
```
wget '--header=Authorization: token ***' -O /home/runner/work/Boomi_TEMP/Boomi_TEMP/QA/consolidated_41a23303-d5ff-40b5-8a07-4c86a1c846c2.zip https://raw.githubusercontent.com/ThotadurgamaniTR/Boomi_TEMP/releases/download/tag4/consolidated_41a23303-d5ff-40b5-8a07-4c86a1c846c2.zip
```
The problem you encounter is that the path does not point to the raw data of the file so it cannot be downloaded.
hope it works for you

**Comment by ThotadurgamaniTR** [comment]
This email is from a contractor that provides managed services to 3rd parties, including Thomson Reuters.
Getting the below error,

$ curl -L -H "Authorization: token ***" -H "Accept: application/binary" https://raw.githubusercontent.com/ThotadurgamaniTR/Boomi_TEMP/releases/download/tag4/consolidated_41a23303-d5ff-40b5-8a07-4c86a1c846c2.zip
404: Not Found


From: Tri Pham Minh ***@***.***>
Sent: Saturday, September 16, 2023 2:03 PM
To: community/community ***@***.***>
Cc: Thota, Durgamani (Extranet) ***@***.***>; Author ***@***.***>
Subject: [EXT] Re: [community/community] unable to download the asset using curl or wget command (Discussion #67271)

External Email: Use caution with links and attachments.


You can use this link to try downloading: "https://raw.githubusercontent.com/ThotadurgamaniTR/Boomi_TEMP/releases/download/tag4/consolidated_41a23303-d5ff-40b5-8a07-4c86a1c846c2.zip<https://urldefense.com/v3/__https:/raw.githubusercontent.com/ThotadurgamaniTR/Boomi_TEMP/releases/download/tag4/consolidated_41a23303-d5ff-40b5-8a07-4c86a1c846c2.zip__;!!GFN0sa3rsbfR8OLyAw!d4mWNeY4_fhf32zx3y03wG-cqWzkOLwORxWuYtOLv6FSYkMHvI-7lZB4QiZyyc5Ctlpvj0D5BJY2zR4X15ygt9WWF9-bIvi-sQo$>"
The problem you encounter is that the path does not point to the raw data of the file so it cannot be downloaded

**Reply by d4kw1n** [reply]
I tried the simulation again and everything is ok.
You can check PAT's permissions to see if it has granted enough permissions.
And make sure the url path has been replaced correctly

**Comment by ThotadurgamaniTR** [comment]
This email is from a contractor that provides managed services to 3rd parties, including Thomson Reuters.
See my PAT permission. Even I am not getting. Can we have a session please


***@***.***

From: Tri Pham Minh ***@***.***>
Sent: Saturday, September 16, 2023 2:23 PM
To: community/community ***@***.***>
Cc: Thota, Durgamani (Extranet) ***@***.***>; Author ***@***.***>
Subject: [EXT] Re: [community/community] unable to download the asset using curl or wget command (Discussion #67271)

External Email: Use caution with links and attachments.


I tried the simulation again and everything is ok.
You can check PAT's permissions to see if it has granted enough permissions.
And make sure the url path has been replaced correctly

**Reply by d4kw1n** [reply]
You can view PAT permissions by visiting:
Settings -> Developer Settings -> Personal Access Tokens -> Tokens ->
Select the token you are using, make sure `repo   Full control of private repositories` is checked.
If you don't have one, you can create a new PAT.

**Comment by ThotadurgamaniTR** [comment]
This email is from a contractor that provides managed services to 3rd parties, including Thomson Reuters.
Yes . below screenshot is that only even iam not able to get

From: Tri Pham Minh ***@***.***>
Sent: Saturday, September 16, 2023 2:38 PM
To: community/community ***@***.***>
Cc: Thota, Durgamani (Extranet) ***@***.***>; Author ***@***.***>
Subject: [EXT] Re: [community/community] unable to download the asset using curl or wget command (Discussion #67271)

External Email: Use caution with links and attachments.


You can view PAT permissions by visiting:
Settings -> Developer Settings -> Personal Access Tokens -> Tokens ->
Select the token you are using, make sure repo Full control of private repositories is checked.
If you don't have one, you can create a new PAT.

**Reply by d4kw1n** [reply]
I describe it with pictures as follows:
![image](https://github.com/community/community/assets/89400593/293d2efa-08c2-434f-8402-9bb0a2590375)
![image](https://github.com/community/community/assets/89400593/7057e729-4fb3-41fe-bede-71076a463a29)
![image](https://github.com/community/community/assets/89400593/d5e7a20a-ea75-471a-9edf-4cbf9e137cd7)
![image](https://github.com/community/community/assets/89400593/a0b7e364-99a6-465d-9df5-85775ab6f980)
![image](https://github.com/community/community/assets/89400593/db1228e1-fe21-48f5-8c52-53e53baa30fa)
![image](https://github.com/community/community/assets/89400593/b03c92bd-bee0-4b4f-bb61-e49ec2f3269f)

**Comment by ThotadurgamaniTR** [comment]
This email is from a contractor that provides managed services to 3rd parties, including Thomson Reuters.

Yeah what you are saying I understood.tbe same steps I followed for creating personal access token and have give the repo scope as permissions...after using that PAT also I am facing the issue....


Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Tri Pham Minh ***@***.***>
Sent: Saturday, September 16, 2023 2:51:33 PM
To: community/community ***@***.***>
Cc: Thota, Durgamani (Extranet) ***@***.***>; Author ***@***.***>
Subject: [EXT] Re: [community/community] unable to download the asset using curl or wget command (Discussion #67271)

External Email: Use caution with links and attachments.


I describe it with pictures as follows:
[image]<https://urldefense.com/v3/__https://user-images.githubusercontent.com/89400593/268439950-293d2efa-08c2-434f-8402-9bb0a2590375.png__;!!GFN0sa3rsbfR8OLyAw!Z3uSwigPBW2tEcA_Nj0DdV4hLpLoXOrMLgJrbffy-72zpf9hGvyddQtBVfhyBbzENcu3US4aoFFDGz9FciNkNTJ9GaLuZdCwwMc$>
[image]<https://urldefense.com/v3/__https://user-images.githubusercontent.com/89400593/268439989-7057e729-4fb3-41fe-bede-71076a463a29.png__;!!GFN0sa3rsbfR8OLyAw!Z3uSwigPBW2tEcA_Nj0DdV4hLpLoXOrMLgJrbffy-72zpf9hGvyddQtBVfhyBbzENcu3US4aoFFDGz9FciNkNTJ9GaLuH0uxyb0$>
[image]<https://urldefense.com/v3/__https://user-images.githubusercontent.com/89400593/268440010-d5e7a20a-ea75-471a-9edf-4cbf9e137cd7.png__;!!GFN0sa3rsbfR8OLyAw!Z3uSwigPBW2tEcA_Nj0DdV4hLpLoXOrMLgJrbffy-72zpf9hGvyddQtBVfhyBbzENcu3US4aoFFDGz9FciNkNTJ9GaLuVLttZsI$>
[image]<https://urldefense.com/v3/__https://user-images.githubusercontent.com/89400593/268440068-a0b7e364-99a6-465d-9df5-85775ab6f980.png__;!!GFN0sa3rsbfR8OLyAw!Z3uSwigPBW2tEcA_Nj0DdV4hLpLoXOrMLgJrbffy-72zpf9hGvyddQtBVfhyBbzENcu3US4aoFFDGz9FciNkNTJ9GaLuAhmGYXc$>
[image]<https://urldefense.com/v3/__https://user-images.githubusercontent.com/89400593/268440090-db1228e1-fe21-48f5-8c52-53e53baa30fa.png__;!!GFN0sa3rsbfR8OLyAw!Z3uSwigPBW2tEcA_Nj0DdV4hLpLoXOrMLgJrbffy-72zpf9hGvyddQtBVfhyBbzENcu3US4aoFFDGz9FciNkNTJ9GaLumYV4sy0$>
[image]<https://urldefense.com/v3/__https://user-images.githubusercontent.com/89400593/268440097-b03c92bd-bee0-4b4f-bb61-e49ec2f3269f.png__;!!GFN0sa3rsbfR8OLyAw!Z3uSwigPBW2tEcA_Nj0DdV4hLpLoXOrMLgJrbffy-72zpf9hGvyddQtBVfhyBbzENcu3US4aoFFDGz9FciNkNTJ9GaLu3622YNQ$>

**Reply by d4kw1n** [reply]
Do you use ultraviewer? If so, can I connect and take a look?

**Comment by ThotadurgamaniTR** [comment]
This email is from a contractor that provides managed services to 3rd parties, including Thomson Reuters.

No we can connect through zoom link if you have please

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Tri Pham Minh ***@***.***>
Sent: Saturday, September 16, 2023 3:05:49 PM
To: community/community ***@***.***>
Cc: Thota, Durgamani (Extranet) ***@***.***>; Author ***@***.***>
Subject: [EXT] Re: [community/community] unable to download the asset using curl or wget command (Discussion #67271)

External Email: Use caution with links and attachments.


Do you use ultraviewer? If so, can I connect and take a look?

**Comment by ThotadurgamaniTR** [comment]
This email is from a contractor that provides managed services to 3rd parties, including Thomson Reuters.


Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by emmett08** [comment]
You will need to get the asset URL first and then use that to download zip. N.B. take note of headers for each.
```
resource "null_resource" "download_github_release" {
  provisioner "local-exec" {
    command = <<-EOT
      if [[ "${var.enabled}" == "true" ]]; then
        asset_url=$(curl -L \
          -H "Accept: application/vnd.github+json" \
          -H "Authorization: Bearer ${var.github_token}" \
          -H "X-GitHub-Api-Version: 2022-11-28" \
          https://api.github.com/repos/${var.github_repo}/releases/latest | jq -r '.assets[] | select(.name == "${var.app_name}-${var.function_name}-${var.sem_ver}.zip").url')
        
        if [[ ! -z "$asset_url" ]]; then
          curl -L -o "${var.app_name}-${var.function_name}-${var.sem_ver}.zip" \
            -H "Accept: application/octet-stream" \
            -H "Authorization: Bearer ${var.github_token}" \
            "$asset_url"
        else
          echo "Asset URL not found"
        fi
      fi
    EOT
  }

  triggers = {
    version = var.sem_ver
  }
}
```
